### PR TITLE
[rllib] Flip sign of A2C, IMPALA entropy coefficient; raise DeprecationWarning if negative

### PR DESCRIPTION
**Pull request by ericl**
## What do these changes do?

Throw an error if we end up penalizing entropy, which is probably unintended. Other options here:
1) break backwards compat by renaming the config
2) always take the absolute value here?

## Related issue number

Closes https://github.com/ray-project/ray/issues/4369

### CODE REVIEW

File: python/ray/rllib/agents/ppo/ppo.py [hunk 1]
```diff
@@ -150,6 +150,8 @@ def update(pi, pi_id):
         return res
 
     def _validate_config(self):
+        if self.config["entropy_coeff"] < 0:
```
Comment (hartikainen) [inline python/ray/rllib/agents/ppo/ppo.py]:
```suggestion
        if self.config["entropy_coeff"] <= 0:
```
Or alternatively change the error blow to "must be non-negative".

Comment (ericl) [reply python/ray/rllib/agents/ppo/ppo.py]: Fixed

**Review by hartikainen — APPROVED** [review]

**Comment by ericl** [comment]
Upon further thought, we can actually safely flip the sign, by raising DeprecationWarning if it's negative.

This is probably the better long-term solution, but should be reviewed carefully.

File: python/ray/rllib/agents/impala/vtrace_policy_graph.py [hunk 2]
```diff
@@ -35,7 +35,7 @@ def __init__(self,
                  bootstrap_value,
                  valid_mask,
                  vf_loss_coeff=0.5,
-                 entropy_coeff=-0.01,
+                 entropy_coeff=0.01,
```
Comment (hartikainen) [inline python/ray/rllib/agents/impala/vtrace_policy_graph.py]: This one is missing a warning.

Comment (hartikainen) [reply python/ray/rllib/agents/impala/vtrace_policy_graph.py]: Same for appo.

Comment (ericl) [reply python/ray/rllib/agents/impala/vtrace_policy_graph.py]: I think the change in impala.py covers both.

**Comment by ericl** [comment]
jenkins retest this please

**Comment by ericl** [comment]
jenkins retest this please

**Pull request merged**